test/common/filter/tcp_proxy_test.cc

```diff
@@ -335,36 +345,36 @@ TEST(TcpProxyConfigTest, EmptyRouteConfig) {
   Json::ObjectSharedPtr json_config = Json::Factory::loadFromString(json);
   NiceMock<Server::Configuration::MockFactoryContext> factory_context_;
 
-  TcpProxyConfig config_obj(*json_config, factory_context_);
+  TcpProxyConfig config_obj(constructTcpProxyConfigFromJson(*json_config, factory_context_));
 
   NiceMock<Network::MockConnection> connection;
   EXPECT_EQ(std::string(""), config_obj.getRouteFromEntries(connection));
 }
 
 TEST(TcpProxyConfigTest, AccessLogConfig) {
-  std::string json = R"EOF(
-    {
-      "stat_prefix": "name",
-      "route_config": {
-        "routes": [
-        ]
-      },
-      "access_log": [
-        {
-          "path": "some_path",
-          "format": "the format specifier"
-        },
-        {
-          "path": "another path"
-        }
-      ]
-    }
-    )EOF";
+  envoy::api::v2::filter::network::TcpProxy config;
+  envoy::api::v2::filter::AccessLog* log = config.mutable_access_log()->Add();
+  log->set_name(Config::AccessLogNames::get().FILE);
+  {
+    envoy::api::v2::filter::FileAccessLog file_access_log;
+    file_access_log.set_path("some_path");
+    file_access_log.set_format("the format specifier");
+    ProtobufWkt::Struct* custom_config = log->mutable_config();
+    MessageUtil::jsonConvert(file_access_log, *custom_config);
```

Comment:
Is this the right way to convert from a specific protobuf config onto a protobuf "Struct"?  I tried making this test use the v2 config directly instead of json conversion, for more diversity of coverage.  But this part seemed really strange to me.

Reply:
Yes, this is the "right" way. Stay tuned for my Any vs. Struct blog post where I will cover this, thanks for reminding me.